ng/src/web/pages/portlists/component.js

```diff
@@ -146,6 +146,105 @@ class PortListComponent extends React.Component {
     return gmp.portlist.import(data).then(onImported, onImportError);
   }
 
+  handleSavePortList(save, data) {
+    const created_port_ranges_copy = [...this.created_port_ranges];
+
+    let promises = created_port_ranges_copy.map(range => {
+      const saveData = {
+        ...range,
+        port_range_start: range.start,
+        port_range_end: range.end,
+        port_type: range.protocol_type,
+      };
+      return this.handleSavePortRange(saveData).then(id => {
+        range.isTmp = false;
+        range.id = id;
+        this.created_port_ranges = this.created_port_ranges
+          .filter(prange => prange !== range);
+        }
+      );
+    });
+    const deleted_port_ranges_copy = [...this.deleted_port_ranges];
+    promises = [...promises, ...deleted_port_ranges_copy.map(range =>
+      this.handleDeletePortRange(range).then(
+        this.deleted_port_ranges = this.deleted_port_ranges
+          .filter(prange => prange !== range)
+      )
+    )];
+    return Promise.all(promises).then(() => save(data));
+  }
+
+  handleTmpAddPortRange(values) {
+    const {tmp_port_ranges} = this.state;
+    const {
+      port_range_end,
+      port_range_start,
+      port_type,
+    } = values;
+
+    // reject port ranges with missing values
+    if (!port_range_start || !port_range_end) {
+      return Promise.reject(new Error(_('The port range needs numerical ' +
```

Comment:
I am not 100% sure if Promise.reject is necessary. It could be possible that `throw new  Error(...) ` is enough.

Reply:
I tried throwing a new Error before, but that didn't work, so it seems we need to catch the rejected promise.

Reply:
Just did a quick look at the corresponding SaveDialog code. Indeed return Promise.reject(new Error is necessary to work as expected.